modules/sdk/src/main/scala/org/tessellation/sdk/domain/snapshot/services/GlobalL0Service.scala

```diff
@@ -37,12 +48,31 @@ object GlobalL0Service {
     l0GlobalSnapshotClient: L0GlobalSnapshotClient[F],
     globalL0ClusterStorage: L0ClusterStorage[F],
     lastGlobalSnapshotStorage: LastSnapshotStorage[F, GlobalIncrementalSnapshot, GlobalSnapshotInfo],
-    singlePullLimit: Option[PosLong]
+    singlePullLimit: Option[PosLong],
+    maybeRealignPeerIds: Option[NonEmptySet[PeerId]] = none
   ): GlobalL0Service[F] =
     new GlobalL0Service[F] {
-
       private val logger = Slf4jLogger.getLogger[F]
 
+      def realignToMajority(lastSnapshot: Hashed[GlobalIncrementalSnapshot]): F[Unit] =
+        maybeRealignPeerIds.traverse_ { realignPeers =>
+          for {
+            peers <- realignPeers.toNonEmptyList
+              .traverse(globalL0ClusterStorage.getPeer)
+              .map(_.toList.flatten)
```

Comment:
I wonder if we should think of some corner cases. The `realignPeers` are decoupled from the `globalL0ClusterStorage` so local reasoning suggests that there could be a situation where non of the nodes are present or e.g. only one is - we will check majority based on this single node. If such a situation should not be possible, maybe it's still better to locally treat such condition as a failure.

Reply:
We would need to have some threshold based on the size of the list, which isn't perfect.

Reply:
See earlier comments: summary: filter failures, apply majority tie breaker based on id sort, if NO replies, continue without check.

Reply:
I don't agree with "continue without check" part and here is a more evolved answer why: https://github.com/Constellation-Labs/tessellation/pull/676#discussion_r1281549837

Reply:
Summary from discussion -- we can't even do this without retries (which aren't implemented,), so I agree with you we have to fail hard here.